backend/app/Actions/Visitors/BounceRateRequest.php

```diff
@@ -0,0 +1,32 @@
+<?php
+declare(strict_types=1);
+
+namespace App\Actions\Visitors;
+
+use App\Http\Requests\Visitors\VisitorsBounceRateHttpRequest;
+use App\Model\Visitors\VisitorsBounceRateFilterData;
+use App\Utils\DatePeriod;
+
+final class BounceRateRequest
+{
+    private $filterData;
+
+    private function __construct(VisitorsBounceRateFilterData $filterData)
+    {
+        $this->filterData = $filterData;
+    }
+
+    public function getFilterData(): VisitorsBounceRateFilterData
+    {
+        return $this->filterData;
+    }
+
+    public static function fromRequest(VisitorsBounceRateHttpRequest $request): self
+    {
+        $period = DatePeriod::createFromTimestamp(
+            $request->getStartDate(),
+            $request->getEndDate()
+        );
+        return new static(new VisitorsBounceRateFilterData($period, $request->getTimeFrame()));
```

Comment:
Not exactly. Your DTO should transfer data, not filters. It means, your DTO should transfer $startDate, $endDate, and $period. Then according to your DTO you should create filter VisitorsBounceRateFilterData inside action, and pass this filter to repository.